packages/neon_framework/lib/src/blocs/apps.dart

```diff
@@ -74,7 +74,19 @@ class _AppsBloc extends InteractiveBloc implements AppsBloc {
     required this.allAppImplementations,
   }) {
     apps.listen((result) {
-      appImplementations.add(result.transform((data) => filteredAppImplementations(data.map((a) => a.id))));
+      appImplementations.add(
+        result.transform(
+          (data) => BuiltSet<AppImplementation>(
+            data
+                .map(
+                  (app) => allAppImplementations.firstWhereOrNull(
+                    (a) => a.id == app.id || (a.additionalMatchingIDs?.contains(app.id) ?? false),
+                  ),
+                )
+                .whereNotNull(),
+          ),
```

Comment:
Please manually loop over `data` and conditionally add the AppImplementation to the set.